src/textual/dom.py

```diff
@@ -1276,17 +1275,27 @@ def toggle_class(self, *class_names: str) -> Self:
         self._update_styles()
         return self
 
-    def has_pseudo_class(self, *class_names: str) -> bool:
-        """Check for pseudo classes (such as hover, focus etc)
+    def has_pseudo_class(self, class_name: str) -> bool:
+        """Check the node has the given pseudo class.
 
         Args:
-            *class_names: The pseudo classes to check for.
+            class_name: The pseudo classes to check for.
 
         Returns:
             `True` if the DOM node has those pseudo classes, `False` if not.
         """
-        has_pseudo_classes = self.pseudo_classes.issuperset(class_names)
-        return has_pseudo_classes
+        return class_name in self.get_pseudo_classes()
+
+    def has_pseudo_classes(self, class_names: set[str]) -> bool:
+        """Check the node has all the given pseudo classes.
+
+        Args:
+            class_names: Set of class names to check for.
+
+        Returns:
+            `True` if all pseudo class names are present.
+        """
+        return class_names.issubset(self.get_pseudo_classes())
```

Comment:
Same as `class_names <= self.get_pseudo_classes()`.
I think it's more elegant, but I'm not going to push too hard for it.

Reply:
The result of `get_pseudo_classes` is not a set, so the operator won't work here.

Reply:
🤦  I left this comment before realising it was a generator.